src/core/gui/toolbarMenubar/ToolMenuHandler.cpp

```diff
@@ -610,6 +610,8 @@ auto ToolMenuHandler::isColorInUse(Color color) -> bool {
 
 auto ToolMenuHandler::getToolItems() -> std::vector<AbstractToolItem*>* { return &this->toolItems; }
 
+auto ToolMenuHandler::getColorToolItems() const -> const std::vector<ColorToolItem*>* { return &this->toolbarColorItems; }
```

Comment:
I think this should return a reference rather than a pointer:
```suggestion
auto ToolMenuHandler::getColorToolItems() const -> const std::vector<ColorToolItem*>& { return this->toolbarColorItems; }
```
(Of course, this also means adapting the calling function)